 drivers/usb/serial/option.c | 12 +++
 1 file changed, 12 insertions(+)

--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -2256,6 +2256,41 @@
 	{ USB_DEVICE_INTERFACE_CLASS(0x2cb7, 0x01a2, 0xff) },			/* Fibocom FM101-GL (laptop MBIM) */
 	{ USB_DEVICE_INTERFACE_CLASS(0x2cb7, 0x01a4, 0xff),			/* Fibocom FM101-GL (laptop MBIM) */
 	  .driver_info = RSVD(4) },
+	{ USB_DEVICE(0x0e8d, 0x7127),						/* Fibocom FM350-GL (RNDIS) */
+	 .driver_info = RSVD(0) | RSVD(1) | RSVD(2)| RSVD(3)| RSVD(4) | RSVD(5) | RSVD(7) | RSVD(8) | RSVD(9) },
+	{ USB_DEVICE(0x0e8d, 0x7126),						/* Fibocom FM350-GL (RNDIS) */
+	 .driver_info = RSVD(0) | RSVD(1) | RSVD(2)| RSVD(3)| RSVD(4) | RSVD(5) | RSVD(7) | RSVD(8) | RSVD(9) },
+	{ USB_DEVICE(0x2cb7, 0x0a04),
+	  .driver_info = RSVD(0) | RSVD(1) | RSVD(5) },				/* Fibocom FM650 ECM */
+	{ USB_DEVICE(0x2cb7, 0x0a05),
+	  .driver_info = RSVD(0) | RSVD(1) | RSVD(6) },				/* Fibocom FM650 NCM */
+	{ USB_DEVICE(0x2cb7, 0x0a06),
+	  .driver_info = RSVD(0) | RSVD(1) | RSVD(6) },				/* Fibocom FM650 RNDIS */
+	{ USB_DEVICE(0x2cb7, 0x0a07),
+	  .driver_info = RSVD(0) | RSVD(1)  | RSVD(6) },				/* Fibocom FM650 MBIM */
+	{ USB_DEVICE(0x2cb7, 0x0110),
+	  .driver_info = RSVD(0) | RSVD(1) },						/* Fibocom FM150 MBIM (29) */
+	{ USB_DEVICE(0x2cb7, 0x0109),
+	  .driver_info = RSVD(2) },									/* Fibocom FM150 RMNET (22) */
+	{ USB_DEVICE(0x2cb7, 0x0105),
+	  .driver_info = RSVD(4) | RSVD(5) | RSVD(6) },				/* Fibocom FM150 ECM (18) & ADB */
+	{ USB_DEVICE(0x2cb7, 0x0104),
+	  .driver_info = RSVD(4) | RSVD(5) | RSVD(6) },				/* Fibocom FM150 RMNET (17) & ADB */
+	{ USB_DEVICE(0x305a, 0x1406),
+	  .driver_info = RSVD(3) | RSVD(4) | RSVD(5) },				/* GXX GM800 ECM */
+	{ USB_DEVICE(0x05c6, 0x90db),
+	  .driver_info = RSVD(2) | RSVD(5) },				/* GXX GM800 RMNET DIAG+AT+RMNET+DPL+QDSS+ADB */
+	{ USB_DEVICE(0x305a, 0x1403),
+	  .driver_info = RSVD(3) | RSVD(4) },				/* GXX GM800 RMNET DIAG+AT+MODEM+RMNET+ADB */
+	{ USB_DEVICE(0x305a, 0x1421),
+	  .driver_info = RSVD(3) | RSVD(4) },				/* GXX GM800 RMNET DIAG+AT+MODEM+RMNET+ADB */
+	{ USB_DEVICE(0x305a, 0x1404),
+	  .driver_info = RSVD(3) | RSVD(4) | RSVD(5) },				/* GXX GM800 RNDIS DIAG+AT+MODEM+RNDIS+ADB */
+	{ USB_DEVICE(0x305a, 0x1405),
+	  .driver_info = RSVD(3) | RSVD(4) },				/* GXX GM800 MBIM DIAG+AT+MODEM+MBIM+ADB */
+	{ USB_DEVICE(0x305a, 0x1402) },				/* GXX GM800 DOWNLOAD MODE DIAG+AT+MODEM */
+	{ USB_DEVICE(0x1c9e, 0x9b05),
+	  .driver_info = RSVD(4) },				/* U8300W */
 	{ USB_DEVICE_INTERFACE_CLASS(0x2df3, 0x9d03, 0xff) },			/* LongSung M5710 */
 	{ USB_DEVICE_INTERFACE_CLASS(0x305a, 0x1404, 0xff) },			/* GosunCn GM500 RNDIS */
 	{ USB_DEVICE_INTERFACE_CLASS(0x305a, 0x1405, 0xff) },			/* GosunCn GM500 MBIM */
